Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM golang:1.16.3-alpine as builder
+
+RUN apk add --update --no-cache make git
+
+WORKDIR /go/src/terragrunt
+
+# Download modules in a separate layer so docker caches it, to reduce build time
+# when modules have not changed.
+COPY go.mod .
+COPY go.sum .
+RUN go mod download -x
+
+COPY . .
+
+RUN make build
+
+###
+
+FROM alpine:latest
```

Comment:
NIT: couldn't this be `scratch`? I think a standalone Go binary doesn't need anything else...

Reply:
Perhaps? Using `alpine:latest` and the multistage build, it's only a 40MB image. Might be over-optimizing... I'll give it a shot though.

Reply:
The scratch base image worked, just needed to use `CGO_ENABLED=0`. Image size is 34MB.

```
$ docker run --rm gruntworkio/terragrunt --version
terragrunt version v0.29.2-4-g4deae14c6db3-dirty
```